jwst/background/background_sub.py

```diff
@@ -26,85 +26,171 @@ def background_sub(input_model, bkg_list, sigma, maxiters):
     bkg_list: filename list
         list of background exposure file names
 
+    sigma: float, optional
+        Number of standard deviations to use for both the lower
+        and upper clipping limits.
+
+    maxiters: int or None, optional
+        Maximum number of sigma-clipping iterations to perform
+
+    bkg_model: JWST data model
+        background data model
+
     Returns
     -------
     result: JWST data model
         background-subtracted target data model
 
     """
+    # Subtract the average background from the member
+    log.debug(' subtracting avg bkg from {}'.format(input_model.meta.filename))
 
     # Compute the average of the background images associated with
     # the target exposure
-    bkg_model = average_background(bkg_list, sigma, maxiters)
+    if input_model.meta.model_type == "ImageModel":
```

Comment:
In fact, now that I think about it more, I'm not sure why `data_dim` is even needed, at least here. It's actually the dimensions (2D vs 3D) of the *background* images that we really care about, as opposed to the dimensions of the science data (input_model). Once the background data have all been averaged together properly, they'll always end up as a 2D image to be subtracted from the input. At that point is doesn't matter what the dimensions of the input are, because you can just rely on numpy array broadcasting to subtract the 2D bkg image from the input, regardless of whether the input is 2D or 3D. Just like we do in the flat_field step where a 2D flat is divided into input data that are either 2D or 3D.  Just use an expression like `input_model.data -= avg_bkg.data` and numpy array broadcasting will do the right thing regardless of whether `input_model.data` is 2D or 3D.

Reply:
Resolved this.